qiskit_experiments/calibration/calibrations.py

```diff
@@ -0,0 +1,698 @@
+        behaviour is needed to handle Call instructions. Each time a Call is found
+        get_parameter_keys is call on the subroutine of the Call instruction and the
+        qubits that are in the subroutine. This also implies carefully extracting the
+        qubits from the subroutine and in the appropriate order.
+
+        Args:
+            schedule: A schedule from which to extract parameters.
+            keys: A set of keys that will be populated.
+            binding_dict: A binding dictionary intended only for channels. This is needed
+                because calling assign_parameters on a schedule with a Call instruction will
+                not assign the parameters in the subroutine of the Call instruction.
+            qubits: The qubits for which we want to have the schedule.
+
+        Returns:
+            keys: The set of keys populated with schedule name, parameter name, qubits.
+        """
+
+        # schedule.channels may give the qubits in any order. This order matters. For example,
+        # the parameter ('cr', 'amp', (2, 3)) is not the same as ('cr', 'amp', (3, 2)).
+        # Furthermore, as we call subroutines the list of qubits involved might shrink. For
+        # example, part of a cross-resonance schedule might involve.
+        #
+        # pulse.call(xp)
+        # ...
+        # pulse.play(GaussianSquare(...), ControlChannel(X))
+        #
+        # Here, the call instruction might, e.g., only involve qubit 2 while the play instruction
+        # will apply to qubits (2, 3).
+
+        qubit_set = set()
+        for chan in schedule.channels:
+            if isinstance(chan, DriveChannel):
+                qubit_set.add(chan.index)
+
+        qubits_ = tuple(qubit for qubit in qubits if qubit in qubit_set)
+
+        for _, inst in schedule.instructions:
+
+            if isinstance(inst, Play):
+                for params in inst.pulse.parameters.values():
+                    if isinstance(params, ParameterExpression):
+                        for param in params.parameters:
+                            keys.add(ParameterKey(schedule.name, param.name, qubits_))
+
+            if isinstance(inst, (ShiftPhase, SetPhase)):
+                if isinstance(inst.phase, ParameterExpression):
+                    for param in inst.phase.parameters:
+                        keys.add(ParameterKey(schedule.name, param.name, (inst.channel.index,)))
+
+            if isinstance(inst, (ShiftFrequency, SetFrequency)):
+                if isinstance(inst.frequency, ParameterExpression):
+                    for param in inst.frequency.parameters:
+                        keys.add(ParameterKey(schedule.name, param.name, (inst.channel.index,)))
+
+            if isinstance(inst, Call):
+                sched_ = inst.subroutine.assign_parameters(binding_dict, inplace=False)
+                keys = Calibrations.get_parameter_keys(sched_, keys, binding_dict, qubits_)
+
+        return keys
+
+    def _get_circuit(
```

Comment:
Seems like you dismiss my comment? This is bit strange because this method is protected but no other methods in this class call this, indicating some external code should access to the protected method. I think generation of `QuantumCircuit` should be done by experiment generators, not by the calibration itself, though this is very convenient. This can be a public util function, which can be put in this module.

Reply:
This could be a useful method when we start using this class. For now I removed it. See here: 37a2eaf Let's see later on if we need this.